osu.Game/Graphics/UserInterface/SegmentedGraph.cs

```diff
@@ -154,7 +156,19 @@ private void recalculateSegments()
             segments.Sort();
         }
 
-        private Colour4 getTierColour(int tier) => tier >= 0 ? tierColours[tier] : new Colour4(0, 0, 0, 0);
+        private ColourInfo getSegmentColour(SegmentInfo segment)
+        {
+            var tierColour = segment.Tier >= 0 ? tierColours[segment.Tier] : new Colour4(0, 0, 0, 0);
+            var ourColour = DrawColourInfo.Colour;
+
+            return new ColourInfo
+            {
+                TopLeft = tierColour * Interpolation.ValueAt<Colour4>(segment.Start, ourColour.TopLeft, ourColour.TopRight, 0, 1),
+                TopRight = tierColour * Interpolation.ValueAt<Colour4>(segment.End, ourColour.TopLeft, ourColour.TopRight, 0, 1),
+                BottomLeft = tierColour * Interpolation.ValueAt<Colour4>(segment.Start, ourColour.BottomLeft, ourColour.BottomRight, 0, 1),
+                BottomRight = tierColour * Interpolation.ValueAt<Colour4>(segment.End, ourColour.BottomLeft, ourColour.BottomRight, 0, 1)
+            };
```

Comment:
Similar function (getting a colour quad out of source colour) exists in [TrianglesV2](https://github.com/ppy/osu/blob/761298cd8e86a935487da5ea46882e05a1e636d3/osu.Game/Graphics/Backgrounds/TrianglesV2.cs#L264), probably would be useful to create one in the framework level

Reply:
Sure. I'm looking to get this out the door for a release tomorrow for now though.

Reply:
Here's a simplified version
```suggestion
        private ColourInfo getSegmentColour(SegmentInfo segment)
        {
            var subQuad = new ColourInfo
            {
                TopLeft = DrawColourInfo.Colour.Interpolate(new Vector2(segment.Start, 0f)),
                TopRight = DrawColourInfo.Colour.Interpolate(new Vector2(segment.End, 0f)),
                BottomLeft = DrawColourInfo.Colour.Interpolate(new Vector2(segment.Start, 1f)),
                BottomRight = DrawColourInfo.Colour.Interpolate(new Vector2(segment.End, 1f))
            };

            subQuad.ApplyChild(segment.Tier >= 0 ? tierColours[segment.Tier] : new Colour4(0, 0, 0, 0));

            return subQuad;
        }
```

Reply:
Applied with minor variations, thanks.